### DISCUSSION TITLE
Polygonal drawing

### DISCUSSION THREAD

**Discussion by yk9710** [visgl/react-google-maps#483] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/fab3ae42-081f-4ef7-b7b4-93e80065f00e)
How do you achieve the effect of clicking and dragging a polygon to draw a line in the drawing library? Where is the source code repository address of this drawing library? I want to implement a controllable polygon in React instead of directly using the drawing tools you provide.

**Comment by mrMetalWood** [comment, marked as answer]
Hi, in the example we are only utilizing the drawing tools that the Google Maps API provides (https://developers.google.com/maps/documentation/javascript/examples/drawing-tools). The source code of that is not available openly since it is part of Googles Maps JS API.

An approach could be to create a polygon with `editable: true` and take care of mouse cursor tracking and updating the polygon coordinates whenever the mouse moves with the current mouse position
